### PR TITLE
Lazy loading & concurrent loading limiting for previews

### PR DESCRIPTION
**Pull request by amessinger**
review @xcambar

### CODE REVIEW

**Comment by amessinger** [comment]
question: should this PR really be merged into v1? or should we create a new one within GH?

File: README.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ DragNDoc is JS application to help splitting and ordering document pages in sub
 Try it
 ------
 
-Small PDF splitter application based on DragNDoc splitpdfapp.com http://splitpdfapp.com 
+Small PDF splitter application based on DragNDoc splitpdfapp.com http://splitpdfapp.com
```
Comment (FlogFr) [inline README.rst:14]: I don't see anything on the URL

Comment (amessinger) [reply README.rst:14]: My editor removed a trailing space character on his own. I don't know a thing about this url, the project it is supposed to link to...

**Comment by kintso** [comment]
Could you put the same loader as displayed on http://webmenu.org/apps/splitpdfapp : ) ?

**Comment by amessinger** [comment]
I can't see any loader (and I thought about it but I'm lazy :p )

**Comment by amessinger** [comment]
@kintso Ok, I'm such a good samaritan. Here's your loader.

File: app/application.coffee [hunk 2]
```diff
@@ -11,6 +11,7 @@ DragNDoc = ((Backbone, Marionette) ->
     callback = null
     callbackText = ""
     helpText = ""
+    maxConcurrentLoadingPages = false
```
Comment (xcambar) [inline app/application.coffee]: You can use `Infinity`, it allows to only rely on numbers in your code below.

File: app/models/page.coffee [hunk 3]
```diff
@@ -10,16 +10,28 @@ PageModel = Backbone.Model.extend(
...
+        viewed: false
+        loading: false
+        loaded: false
 )
 
 # Page Collection
 PagesCollection = Backbone.Collection.extend(
+    _toBeLoadedQueue: []
+    _loadingSlotAvailable: ->
+        return not @maxConcurrentLoadingPages or @where({loading: true}).length < @maxConcurrentLoadingPages
```
Comment (xcambar) [inline app/models/page.coffee]: You can remove `not @maxConcurrentLoadingPages or` if you use `Infinity` (see my comment above)

**Review by xcambar — CHANGES_REQUESTED** [review]
Code LGTM, but I still need to test it.

File: app/models/page.coffee (hunk 3 above)
Comment (amessinger) [reply app/models/page.coffee]: True! Thanks :)

**Review by xcambar — APPROVED** [review]

**Comment by xcambar** [comment]
Tests fail because of:

> Could not find .travis.yml, using standard configuration.

Can you fix it?

**Pull request merged**